media-gfx/nomacs/nomacs-3.14.2.ebuild

```diff
@@ -0,0 +1,86 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake xdg
+
+DESCRIPTION="Qt-based image viewer"
+HOMEPAGE="https://nomacs.org/"
+SRC_URI="
+	https://github.com/${PN}/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz
+	plugins? ( https://github.com/${PN}/${PN}-plugins/archive/${PV}.tar.gz -> ${PN}-plugins-$(ver_cut 1-2).tar.gz )
+"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux"
+IUSE="+jpeg +opencv plugins raw tiff zip"
+
+REQUIRED_USE="
+	raw? ( opencv )
+	tiff? ( opencv )
+"
+
+RDEPEND="
+	dev-qt/qtconcurrent:5
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5[jpeg?]
+	dev-qt/qtnetwork:5
+	dev-qt/qtprintsupport:5
+	dev-qt/qtsvg:5
+	dev-qt/qtwidgets:5
+	media-gfx/exiv2:=
+	opencv? ( >=media-libs/opencv-3.4:= )
+	raw? ( media-libs/libraw:= )
+	tiff? (
+		dev-qt/qtimageformats:5
+		media-libs/tiff:0
+	)
+	zip? ( dev-libs/quazip[qt5(+)] )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	dev-qt/linguist-tools:5
+	virtual/pkgconfig
+"
+
+S="${WORKDIR}/${P}/ImageLounge"
+
+DOCS=( src/changelog.txt )
+
+src_unpack() {
+	unpack "${P}.tar.gz"
+	if use plugins ; then
+		einfo "Unpacking ${PN}-plugins-$(ver_cut 1-2).tar.gz to /var/tmp/portage/sci-physics/${P}/work/${P}/ImageLounge/plugins"
+		mkdir "${P}"/ImageLounge/plugins
+		tar -C "${P}"/ImageLounge/plugins --strip-components=1 -xzf "${DISTDIR}/${PN}-plugins-$(ver_cut 1-2).tar.gz" || die
```

Comment:
Don't manually invoke `tar`, see https://devmanual.gentoo.org/ebuild-writing/functions/src_unpack/index.html

Just let portage do the unpacking and then use `mv` if needed

Reply:
This plugin library is fully runtime dependency. It works without re-compilation of nomacs for nomacs-3.12 too and for lximage-qt.

Should I create COMMON_DEPS variable and move to it all stuff without this plugin and then add this new variable to RDEPEND and DEPEND?

Reply:
Replaced with conditional unpack call.

I also fix installation and search nomacs-plugin directory in `src_prepare()` section:
```
if use plugins ; then
	# Fix nomacs-plugins installation and search library directory
	sed -i "s:lib/nomacs-plugins:$(get_libdir)/nomacs-plugins:" "${S}/plugins/cmake/Utils.cmake" || die
	sed -i "s:lib/nomacs-plugins:$(get_libdir)/nomacs-plugins:" "${S}/src/DkCore/DkPluginManager.cpp" || die
fi
```
to avoid installation of plugins into `/usr/lib` for `amd64` arch.

Reply:
Move early `RDEPEND` to `COMMON_DEPEND` variable with placement of `media-plugins/qt-heif-image-plugin` into `RDEPEND` without placement into `DEPEND`.
```
RDEPEND="
	${COMMON_DEPEND}
	heif? ( media-plugins/qt-heif-image-plugin )
"
DEPEND="${COMMON_DEPEND}"
```